### PR TITLE
add support for Ecto.Enum

### LINKED ISSUE #7: support for Ecto.Enum
Definitions like  `field :foo, Ecto.Enum, values: [:bar1, :bar2]` should produce a typespec like `:bar1 | :bar2`

### PR DESCRIPTION
**Pull request by dvic**
Closes #7

### CODE REVIEW

**Comment by dvic** [comment]
Note that this bumps Ecto to version `3.5` (because of `Ecto.Enum`)

**Comment by dvic** [comment]
@bamorim did you have any chance to look at this? :)

**Comment by bamorim** [comment]
Sorry, I was on vacation after quitting my job hahaha.
I'll try to dedicate more effort for this lib. Thanks for the contribution. I'll test this and merge (looks good already).

**Comment by dvic** [comment]
Hope you had a good time, no worries, take your time. Thanks for the library!

File: lib/typed_ecto_schema/ecto_type_mapper.ex [hunk 1]
```diff
@@ -109,6 +114,23 @@ defmodule TypedEctoSchema.EctoTypeMapper do
...
+      unquote(first) | unquote(last)
+    end
+  end
+
+  defp disjunction_typespec([head | tail]) when is_atom(head) do
+    quote do
+      unquote(head) | unquote(disjunction_typespec(tail))
+    end
+  end
+
```
Comment (bamorim) [inline lib/typed_ecto_schema/ecto_type_mapper.ex:133]:
```suggestion

  defp disjunction_typespec(_) do
    quote do
      any()
    end
  end
```

I tested this and the only problem I found was that if the user made a mistake, such as not declaring the values or something around these lines, the code was throwing a cryptic error such as "no pattern match" and was omitting the nicely printed error Ecto gives us:

```
== Compilation error in file lib/test.ex ==
** (ArgumentError) Ecto.Enum types must have a values option specified as a list of atoms. For example:

    field :my_field, Ecto.Enum, values: [:foo, :bar]
```

So I just added this fallback that just defaults to `any()`. It is not great, but makes this lib less intrusive.

But apart from this detail, great job!

**Comment by bamorim** [comment]
@dvic I think my suggestion did not contained a blank line after the function definition. Sorry about that, my bad.

**Pull request merged**